packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -126,11 +116,9 @@ export function nextSibling(node: Node): Node | null {
     return node.nextSibling;
 }
 
-export function attachShadow(element: Element, options: ShadowRootInit): ShadowRoot {
-    if (hydrating) {
-        return element.shadowRoot!;
-    }
-    return element.attachShadow(options);
+export function attachShadow(element: HTMLElement, options: ShadowRootInit): ShadowRoot {
+    const internals = element.attachInternals?.();
```

Comment:
@nolanlawson yes, as @pmdartus pointed out, the problem persist on this line. The fact that tagNames are not registered during rehydration makes this line to throw when `attachInternals` is present, but the element was not upgraded by the UA. I don't have a solution just yet for this problem, in another PR I'm proposing to always register before any form of upgrading, but this PR should get in first... maybe just adding a try/catch for now and eventually removing the try/catch.

Reply:
I have added the try/catch.